ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java

```diff
@@ -513,28 +485,25 @@ public static void slash_validator(BeaconState state, int slashed_index) {
    */
   public static Bytes32 get_block_root(BeaconState state, UnsignedLong epoch)
       throws IllegalArgumentException {
-    return get_block_root_at_slot(state, compute_start_slot_of_epoch(epoch));
+    return get_block_root_at_slot(state, compute_start_slot_at_epoch(epoch));
   }
 
   /**
-   * Return the number of committees at ``epoch``.
+   * return the number of committees at ``slot``.
    *
    * @param state
-   * @param epoch
+   * @param slot
    * @return
-   * @see
-   *     <a>https://github.com/ethereum/eth2.0-specs/blob/v0.8.0/specs/core/0_beacon-chain.md#get_committee_count</a>
    */
-  public static UnsignedLong get_committee_count(BeaconState state, UnsignedLong epoch) {
+  public static UnsignedLong get_committee_count_at_slot(BeaconState state, UnsignedLong slot) {
+    UnsignedLong epoch = compute_epoch_at_slot(slot);
     List<Integer> active_validator_indices = get_active_validator_indices(state, epoch);
-    return max(
-            UnsignedLong.ONE,
-            min(
-                UnsignedLong.valueOf(Math.floorDiv(SHARD_COUNT, SLOTS_PER_EPOCH)),
-                UnsignedLong.valueOf(active_validator_indices.size())
-                    .dividedBy(UnsignedLong.valueOf(SLOTS_PER_EPOCH))
-                    .dividedBy(UnsignedLong.valueOf(TARGET_COMMITTEE_SIZE))))
-        .times(UnsignedLong.valueOf(SLOTS_PER_EPOCH));
+    return UnsignedLong.valueOf(
+        Math.max(
```

Comment:
You seem to be missing the max(1, min(... outer loop here.

Reply:
https://github.com/ethereum/eth2.0-specs/blob/v0.9.0/specs/core/0_beacon-chain.md#get_committee_count_at_slot

Reply:
I made a PR to your repo for this, fyi.